Modal windows, Congratulation, Resend Link

IgorOskin1984 opened the pull request.

IgorOskin1984 committed:
feat, style(EmailSent):
1. create folder email_sent/ui, and files  emailSent, scss, index.
2. add icon
3. initial styles
4. render EmailSent
via src/app/[ lng ]/page.tsx

IgorOskin1984 committed:
style(EmailSent): import reset styles
delete box-sizing: border-box; decaoause it has reset styles

IgorOskin1984 committed:
feat & style(EmailSent):
1. add Link and Button
2. style Button

IgorOskin1984 committed:
fix(EmailSent):
1. Rename files and folders.
2. Create SignUpModalPage

IgorOskin1984 committed:
feat(SignUpModalWindow change language):
1. add data to JSON change language
2. use t from useClientTranslation to change language

IgorOskin1984 committed:
fix(SignUpModalWindow): change directory
add type and props to SignUpModalPage
add change language json`s

IgorOskin1984 committed:
Merge branch 'main' of https://github.com/Invincible-ship/inctagram into signUpModalWindow

IgorOskin1984 committed:
Merge branch 'ui-signUp' of https://github.com/Invincible-ship/inctagram into signUpModalWindow

IgorOskin1984 committed:
merge with main and ui-signUp

IgorOskin1984 committed:
Initial commit: install

IgorOskin1984 committed:
feat ui(AccountMergingPage): create files.
Render on screen

IgorOskin1984 committed:
feat & style (AccountMerging)

IgorOskin1984 committed:
feat(AccountMerging):
1. Отрисовываю разные рисунки в зависимости от сценария
2. Вынес Title в отдельную компоненту

IgorOskin1984 committed:
fix(AccountMerging):
rename files & folders
fix imports

IgorOskin1984 committed:
порсто тест

IgorOskin1984 committed:
feat (SignUpAdditionPages):
1. Создал переводы
2. Исправил стили
3. Скачал фото для Verification
4. Исправил импорты
5. Сделал скилет CommonBlock и начал использовать его вместе с children

IgorOskin1984 committed:
feat(SignUpAdditionPages):
вынес CommonBlock, MergeAccount, Congratulation, Verification и styles
в отдельные файлы.
Пока не отрисовывал

IgorOskin1984 committed:
feat([lng] page):
1.Отрисовал все 3 компоненты на экране
2.Для этого прописал добавлялся в Page,
и index
3. Написал SignUpAdditionPages с тремя компонентами

IgorOskin1984 committed:
styled(  SignUpAdditionPages )

IgorOskin1984 committed:
npm i

IgorOskin1984 committed:
Merge commit 'b1d2c10e503cff77763ee296c87c6837eef2c22d' into signUpModalWindow

IgorOskin1984 committed:
fix (SignUpModalWindow):
rename function, file & folder names

IgorOskin1984 committed:
Перед тем как я узнал, что у нас есть Modal.tsx
1.Style SignUpPage, SignUpModalPage
2.ModalWindow part of logic

IgorOskin1984 committed:
feat( SignUpModalWindow ):
1. SignUp coment errors
2. style SignUpModalWindow
3. Make ModalWindow
  - use Modal component
4. style Modal component

IgorOskin1984 committed:
fix import, rename files and folder

IgorOskin1984 committed:
refactoring ( SignUpModal ):
1. Перенес ModalWindow в отдельный файл папки Modal/children
2. Туда же перенес и переименовал стили.
3. Переименовал SignUpModalWindow в SignUpModal
4. Сделал onClose обязательным в ModalWindowProps

IgorOskin1984 committed:
add 'use client'

IgorOskin1984 committed:
Сохранился перед удалением
node_modules, файл package.json и package-lock.json yarn.lock

IgorOskin1984 committed:
Удалаил все ниже перечисленное

IgorOskin1984 committed:
npm i

IgorOskin1984 committed:
instal axios

IgorOskin1984 committed:
install @reduxjs/toolkit

IgorOskin1984 committed:
install react-hook-form

IgorOskin1984 committed:
install zod

IgorOskin1984 committed:
npm i pacage json from signup

IgorOskin1984 committed:
Попытки отобразить и потестить svg
Он почему-то не видит локальные svg, даже из public.
jest из коробки не работает с svg
Абзац

IgorOskin1984 committed:
Merge commit 'aa7a09fa9fd38f2e869cefab904ce863aa9654c5' into signUpModalWindow
SignUp: conflicts

IgorOskin1984 committed:
npm i

IgorOskin1984 committed:
feat( SignUp):
1. SignUp: isOpen, userEmail, onClose, onSubmit
2.SignUpModal props
3. ModalWindow types, {userEmail && ...}
4. ModalWindowBody - тело запроса убрал отдельно

IgorOskin1984 committed:
coment

IgorOskin1984 committed:
Merge commit
'7b8c51d6b912cc2c61c67b4e4273b509af99356e' into signUpModalWindow

IgorOskin1984 committed:
отрисовался в браузере

IgorOskin1984 committed:
fix ( CongratulationPage):
1. create routting folder
2. fixing props for page-additionalComponents
RESULT - rout is work

IgorOskin1984 committed:
feat (Congratulation):
1. take query from useSearchParams
2. in useEffect
 - get code from query.get('confirmationCode')
  - set code to state

IgorOskin1984 committed:
feat (Congratulation)
redirect to login

IgorOskin1984 committed:
fix ( .... ):
1. Роут для MergeAccountPage
2. regisrationAPI: создал запросы для SignUp  в файле rtkQslice.ts
3. Подключил regisrationAPI к store, добавил middleware
4. Отправил данные на сервер.
5. Email из ответа передал в SignUpModal.
6. Бдагодаря response.isLoading убрал лишнее из ModalWindow

IgorOskin1984 committed:
feat( regisrationAPI, Congratulation ):
registrationConfirmation
Congratulation take confirmationCode from URL and send it to server

IgorOskin1984 committed:
feat & refactor
1. Создал EmailConfirmation где отображаю или Congratulation
или Verification в зависимости от ответа сервера.
Тут же  хочу отобразить модалку
2. Испраил импорты.
3. Удалил за ненадобностью SignUpAdditionPages.
Надеюсь, что это не ошибка.
4. Убрал логику из Congratulation
5.ПЕРЕДАЛ на сервер confirmationCode через квери параметры урла

IgorOskin1984 committed:
feat ( regisrationAPI ):
1/ emailResending request
2/ get resendLinkToEmail from
useEmailResendingMutation in EmailConfirmation
3/ refactoring types and code

IgorOskin1984 committed:
refactor :
routes congratulation & resendLink

IgorOskin1984 committed:
refactor (EmailConfirmation):
1/ take status from url query
2/ if status === emailIsConfirmed render Congratulation
3/ if (status === emailWasUsed) render Congratulation & Modal

IgorOskin1984 committed:
feat(EmailConfirmationModal):
 create EmailConfirmationModal

IgorOskin1984 committed:
feat & refactor ( EmailConfirmationModal ) :
1. перенес логику из ModalWindow в EmailConfirmationModal
2. Исправил типы в EmailConfirmationModalPropsType
для EmailConfirmationModal
3. Добавил переводы
4. Убрал лишний код и обязательные типы

IgorOskin1984 committed:
refactor (EmailConfirmationModal )
1/ delete useState
2 edit props

IgorOskin1984 committed:
Merge branch 'EmailConfirmationModal' into registration-confirmation

IgorOskin1984 committed:
refactor (routes, types ) :
1/ rename rout-folders & anoter folders, files and functions
2/Make ResendLinkModal
3/ put sum types at the and of page

IgorOskin1984 committed:
feat (ResendLink) :
 POST request to /auth/confirmation-email-resending
1/take email from url
2/send it via sendLinkAgain from useEmailResendingMutation

IgorOskin1984 committed:
feat (ResendLink):
1/{ isSuccess && setIsOpen(true) }
2/onClose
3/if (isLoading)
4/valid props to ResendLinkModal
5/ consts for tranlation

IgorOskin1984 committed:
refactor( language change)
1/ all files and functions in lowercase
2/ all components have consts for language switching

IgorOskin1984 committed:
refactor (ResendLinkModal): add userEmail props

IgorOskin1984 committed:
Merge branch 'resendLink' into registration-confirmation

IgorOskin1984 committed:
refactor rout name

IgorOskin1984 committed:
Merge branch 'registration-confirmation' into signUpModalWindow

IgorOskin1984 committed:
feat ( ErrorModal ): create, add language change

IgorOskin1984 committed:
Merge commit 'c6b0b2cd58bc4503cdaa981a9617c3d5d1e8b115' into signUpModalWindow

IgorOskin1984 committed:
install after merge

IgorOskin1984 committed:
test (ModalWindow) : chat GPT is the best))

IgorOskin1984 committed:
Merge branch 'main' into signUpModalWindow

IgorOskin1984 committed:
npm i

IgorOskin1984 committed:
fix after merfe
 another files was added too

IgorOskin1984 committed:
refacor ( signUpApi ) :
add endpoints & emailResending
fix import in ResendLink

IgorOskin1984 committed:
fix (ModalWindow Test) after eslint
LF / CRLF???

IgorOskin1984 committed:
Merge branch 'main' into signUpModalWindow

IgorOskin1984 committed:
fix after merge

IgorOskin1984 committed:
fix:
В папке providers удалил папку config и файл StoreProvider.tsx

IgorOskin1984 committed:
fix (ResendLink API & language change):
1/ create resendLinkApi
2/ fixt modal window leverage
3/ fixt rtkApi import

IgorOskin1984 committed:
fix & refactor & language switching
1 modal style column-gap
2 ModalWindow types without lng: string;
3 Types for ResendLinkModal, ErrorModal, EmailConfirmationModal
4 ErrorModal with userEmail and without it
5 ResendLink:
 - if noEmail
 - language switching was fix(mostly by it self/ Mybe after merge)
6 delete signUpApi
7 add transation

IgorOskin1984 committed:
refactor ResendLink and Congratulation
arhitectoring changes

IgorOskin1984 committed:
refactor code CongratulationPage ResendLinkPage

IgorOskin1984 committed:
refacor congratResendMerge
1 renamed som files and folders
2 deleted some folders
3 rename some types

IgorOskin1984 committed:
refactor
1 congratResendMergePropsType
2 CongratResendUIPropsType

IgorOskin1984 committed:
refactor ModalWindows types

IgorOskin1984 committed:
fix by lint:ts:fix
& fix ModalWindow.test  rest props

IgorOskin1984 committed:
refactor ModalWindow path

IgorOskin1984 committed:
Merge branch 'main' into signUpModalWindow
1 merge
2 install
3 useDispatch from react-redux
4 create deleted useModal hook
create deleted useModal hook

IgorOskin1984 committed:
fix after merge

IgorOskin1984 committed:
refactor:
1 deleted signIn
2 created Close component

IgorOskin1984 committed:
stories for ModalWindow

IgorOskin1984 committed:
refactor types, ModalWindow Story

IgorOskin1984 committed:
feat stories
1 CongratulationModal
2 Modal
3 refactor ModalWindowStiory
4 ModalWindow add data-testid
5 install storybook/jest

IgorOskin1984 committed:
feat & refactor
1 CongratulationUiStory
2 create and use PictureCongratulation component

IgorOskin1984 committed:
refactor Modal & ModalWindow stories
&& fix type ModalWindow

IgorOskin1984 committed:
testing:
  stories create
  - ErrorModalStory
  - ResendLinkModalStory
  - ResendLinkUiStory
  stories refactor
  - ModalWindow
  refactor code
  - create PictureVerification

IgorOskin1984 committed:
refactor Congratulation
 - add ErrorModal
 - lint:ts:fix

IgorOskin1984 committed:
Merge commit 'd945bb77c30c8956aa8db6866c1e2e93269fac3c' into signUpModalWindow

IgorOskin1984 committed:
Revert "Merge commit 'd945bb77c30c8956aa8db6866c1e2e93269fac3c' into signUpModalWindow"

This reverts commit 16776b4258b7010605836907f7a14257d7e3871d, reversing
changes made to a8d9c8302e48b8749ca8f9d1879480427c16242d.

IgorOskin1984 committed:
Revert "Revert "Merge commit 'd945bb77c30c8956aa8db6866c1e2e93269fac3c' into signUpModalWindow""

This reverts commit 054aa2d3695cf0f18c0b43b3d5dc9f2c9dde925a.

IgorOskin1984 committed:
fix after merge

IgorOskin1984 committed:
Merge branch 'main' into signUpModalWindow
- feat Signup - add Modal
- refacror types

IgorOskin1984 committed:
refactor types

IgorOskin1984 committed:
refactor congratResendMergeFeatures
 - deleted ui folder
 - fix import

IgorOskin1984 committed:
lint:ts:fix

IgorOskin1984 committed:
test:
 - SignUpModal
 - CommonBlock

IgorOskin1984 committed:
tset CongratulationUI && refactor
 type import
 - CongratulationUI
 - ResendLinkUI

IgorOskin1984 committed:
test Modal
 - all the athers modal windows tests was deleted

IgorOskin1984 committed:
test refactor
 - CongratulationUI
 - add svg role = {img} to PictureCongratulation

IgorOskin1984 committed:
test ResendLinkUI
add role='img' for svg

IgorOskin1984 committed:
fix imports

IgorOskin1984 committed:
install chromatic

IgorOskin1984 committed:
Merge branch 'main' into signUpModalWindow

IgorOskin1984 committed:
refactor and fix after merge

IgorOskin1984 committed:
fix: eslint+prettier
 install eslint-config-prettier

IgorOskin1984 committed:
refactor by prettier

vasilec97 commented on client/css/Button.module.css:
Откуда взялся css? Это касается всей папки.

IgorOskin1984 replied:
Понятия не имею, откуда он взялся))
Я нашел тот комит, где он появился. Но почему и как - для меня загадка.
Буду фиксить

vasilec97 commented on client/src/features/auth/signup/model/api/signUpApi.ts:
У фичей нету собственного api, данный endpoint должен быть в userApi. Request Type и Response Type типизируются в дженерике build.mutation<>.

IgorOskin1984 replied:
Исправил.

vasilec97 reviewed (commented):
- Переработать файловую структуру, в соответствии с codestyle проекта (по аналогии с уже существующим кодом);
- Избавиться от папки children в sharedModal. Modal - это универсальная компонента (никаких ModalWindow там быть не должно) и сама компонента ModalWindow бесполезная, она только раздувает код и делает props drilling.
- Большое количество ненужных компонент - избавиться от них. Создай отдельную фичу в features/auth - emailConfiramtion, и там в папке ui напиши страницы для congratulation и resendLink, без кучи ненужных компонент.
- Сделать правки по остальным файлам, в которых есть review

IgorOskin1984 committed:
fixing:
 - css folder was deleted
 - useClientTranslation first argument is "empty string"
 - build.storybook from stash

IgorOskin1984 committed:
fix:
 - API to userApi
 - fix hook import
 - create types for API

IgorOskin1984 committed:
cahge folders

IgorOskin1984 committed:
refactor:
- ResendLink change folder
- ModalWindow refactor
- useModal deleted from Modal

IgorOskin1984 committed:
refactor:
- Congratulation & ResendLink убрад из features в pages
- сделал MainComponent для отрисовки Congratulation & ResendLink,
где собрал его составные части
- сделал EmailConfiramtion,
где написал логику отрисовки Congratulation или ResendLink
- ModalWindow перенес в features/auth/signup/ui/modalWindow,
что бы им можно было пользоваться в pages
- исправил файлы смены языка

IgorOskin1984 committed:
fix
rename EmailConfirmation

IgorOskin1984 committed:
ref & fix
- const-s to lib segment
- fix css imports
- add folder MainComponent, fix import

IgorOskin1984 committed:
test ModalWindow

IgorOskin1984 committed:
lint:ts:fix & storybook

IgorOskin1984 committed:
Merge branch 'fixing-branch' into signUpModalWindow

IgorOskin1984 committed:
ref: Modal test delete

IgorOskin1984 committed:
fix: useClientTranslation
deleted lng

IgorOskin1984 commented:
Закончил. Надеюсь, что все норм

IgorOskin1984 commented:
Нажал "закрыть" теперь нажимаю "открыть" пул реквест

vasilec97 reviewed (commented):
- Разрешить конфликты с основной веткой

IgorOskin1984 committed:
Merge commit '14dcce9f7a6a630c45b2902e522c247aef4c361d'
main into signUpModalWindow

IgorOskin1984 commented:
обновил мейн, сделал мерж, пофиксил конфликты, запустил проект. работает.
проверил свои линтер, тесты, сторис и хроматик.
jest test у меня примитивные. Проверка работы api-хука для emailConfirmation остановилась на уровне, "произошел ли клик по кнопке"

IgorOskin1984 committed:
linnter & fix

IgorOskin1984 committed:
Merge branch 'main' into signUpModalWindow

IgorOskin1984 committed:
npm install & lint:ts:fix

IgorOskin1984 committed:
refactor MainComponent
- storybook main.ts prewiew.tsx
- package.jsin dependevsies for storybook
- i18n.ts
- icons
- MainComponent parts

vasilec97 committed:
add signin with google

vasilec97 committed:
fix google oauth

vasilec97 committed:
finish github/google auth

vasilec97 committed:
fix

vasilec97 committed:
fix

vasilec97 committed:
Merge branch 'third-party-oauth' of https://github.com/Invincible-ship/inctagram into third-party-oauth

vasilec97 committed:
Merge branch 'third-party-oauth' of https://github.com/Invincible-ship/inctagram into third-party-oauth

vasilec97 committed:
Merge branch 'third-party-oauth' of https://github.com/Invincible-ship/inctagram into third-party-oauth

IgorOskin1984 committed:
fix:
- svg,
- MainComponent: useClientTranslation -> namespaces
- stories

IgorOskin1984 committed:
Merge branch 'main' into signUpModalWindow

IgorOskin1984 committed:
refactor: SignUp -> ModalWindow

IgorOskin1984 committed:
test: SignUp ModalWindow tests

IgorOskin1984 committed:
Merge branch 'third-party-oauth' into signUpModalWindow
- npm i
- src/.env
- storybook main @svgr/webpack ???

IgorOskin1984 committed:
src/.env
